packages/typechain-target-ethers/lib/generation.ts

```diff
@@ -242,54 +245,51 @@ function generateEventArgType(eventArg: EventArgDeclaration): string {
 
 // https://docs.ethers.io/ethers.js/html/api-contract.html#types
 function generateInputType(evmType: EvmType): string {
-  switch (evmType.constructor) {
-    case IntegerType:
+  switch (evmType.type) {
+    case "integer":
       return "BigNumberish";
-    case UnsignedIntegerType:
+    case "uinteger":
       return "BigNumberish";
-    case AddressType:
+    case "address":
       return "string";
-    case BytesType:
-    case DynamicBytesType:
+    case "bytes":
+    case "dynamic-bytes":
       return "Arrayish";
-    case ArrayType:
-      return `(${generateInputType((evmType as ArrayType).itemType)})[]`;
-    case BooleanType:
+    case "array":
+      return `(${generateInputType(evmType.itemType)})[]`;
+    case "boolean":
       return "boolean";
-    case StringType:
+    case "string":
       return "string";
-    case TupleType:
-      return generateTupleType(evmType as TupleType, generateInputType);
-
+    case "tuple":
+      return generateTupleType(evmType, generateInputType);
     default:
-      throw new Error(`Unrecognized type ${evmType.constructor.name}`);
+      throw new UnreachableCaseError(evmType);
```

Comment:
is the `default` case even necessary? now that we have an exhaustive switch over a discriminated union, it woudl be even better to remove the `default` case, so the compilation fails if we forget a case above

Reply:
Same question for output type switch below, and for other targets.

Reply:
Oh, actually it's not needed with explcit signature like this 👍 good point